### ISSUE TITLE
ConfigMap "operator-sdk-run-bundle-config" is invalid: []: Too long

### ISSUE THREAD

**Issue by loganmc10** [operator-framework/operator-sdk#5880]
#### What did you do?
Tried updating my operator on the OperatorHub, see https://github.com/k8s-operatorhub/community-operators/pull/1339

#### What did you expect to see?
A successful upgrade

#### What did you see instead? Under which circumstances?
```
[root@routernetes ~]# operator-sdk run bundle quay.io/routernetes/routernetes-operator-bundle:v0.0.3 -n operators
INFO[0010] Creating a File-Based Catalog of the bundle "quay.io/routernetes/routernetes-operator-bundle:v0.0.3" 
INFO[0012] Generated a valid File-Based Catalog         
INFO[0018] Created registry pod: quay-io-routernetes-routernetes-operator-bundle-v0-0-3 
INFO[0018] Created CatalogSource: routernetes-operator-catalog 
INFO[0018] Created Subscription: routernetes-operator-v0-0-3-sub 
INFO[0019] Approved InstallPlan install-2n54p for the Subscription: routernetes-operator-v0-0-3-sub 
INFO[0019] Waiting for ClusterServiceVersion "operators/routernetes-operator.v0.0.3" to reach 'Succeeded' phase 
INFO[0019]   Waiting for ClusterServiceVersion "operators/routernetes-operator.v0.0.3" to appear 
INFO[0030]   Found ClusterServiceVersion "operators/routernetes-operator.v0.0.3" phase: Pending 
INFO[0032]   Found ClusterServiceVersion "operators/routernetes-operator.v0.0.3" phase: InstallReady 
INFO[0033]   Found ClusterServiceVersion "operators/routernetes-operator.v0.0.3" phase: Installing 
INFO[0073]   Found ClusterServiceVersion "operators/routernetes-operator.v0.0.3" phase: Succeeded 
INFO[0073] OLM has successfully installed "routernetes-operator.v0.0.3" 
[root@routernetes ~]# operator-sdk run bundle-upgrade quay.io/routernetes/routernetes-operator-bundle:v0.0.4 -n operators
INFO[0002] Found existing subscription with name routernetes-operator-v0-0-3-sub and namespace operators 
INFO[0002] Found existing catalog source with name routernetes-operator-catalog and namespace operators 
INFO[0014] Generated a valid Upgraded File-Based Catalog 
FATA[0014] Failed to run bundle upgrade: update catalog error: error creating registry: error building registry pod definition: configMap error: error updating ConfigMap: ConfigMap "operator-sdk-run-bundle-config" is invalid: []: Too long: must have at most 1048576 bytes 
```

#### Environment

**Operator type:**
Operator is here: https://github.com/routernetes/routernetes-operator
Ansible based operator that was recently updated to SDK 1.22.0 from 1.21.0

The upgrade was from 0.0.3 (SDK 1.22.0) to 0.0.4 (SDK 1.22.0)

**Kubernetes cluster type:**

`$ operator-sdk version`

`$ go version` (if language is Go)

`$ kubectl version`

**Comment by loganmc10** [comment]
Maybe the same issue as https://github.com/operator-framework/operator-sdk/issues/5870? But this is affecting 1.22.0

**Comment by everettraven** [comment]
@loganmc10 This is something that was an issue before the 1.22.0 release but we weren't able to get the bugfix for `run bundle-upgrade` in on the 1.22.0 release. It is planned that this bug will be fixed and a patch release (1.22.1) will be created once the fix is in.

@rashmigottipati is working on the fixes for this so she may have more information on this.

**Comment by rashmigottipati** [comment]
@loganmc10 I'm able to reproduce this on latest master. We recently got some fixes in for `run bundle` and `run bundle-upgrade` to make them work with a large FBC index. This change resolves most of the scenarios where there is a large FBC index and the rendered bundle image is within `ConfigMap` size limits as we modified the registry pod logic to use a `ConfigMap` to store the extra FBC (which is the rendered bundle). But this will only work if the rendered bundle is within the 1MB, typical size limits of a `ConfigMap`. After doing some debugging, I figured that the image you provided is failing due to the same reason.

We will be looking into solving this particular case next - to resolve it for bundles more than 1MB. I will keep you posted here on any updates on this! Thanks.
